feat: add `stats/strided/nancount` (#11057)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/strided/nancount/README.md

<!--

@license Apache-2.0

Copyright (c) 2026 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# nancount

> Calculate the number of non-`NaN` elements in a strided array.

<section class="intro">

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var nancount = require( '@stdlib/stats/strided/nancount' );
```

#### nancount( N, x, strideX )

> Calculate the number of non-`NaN` elements in a strided array

Reviewer: ```suggestion
Calculates the number of non-`NaN` elements in a strided array.
```

```javascript
var x = [ 1.0, -2.0, NaN, 2.0 ];

var v = nancount( x.length, x, 1 );
// returns 3
```

The function has the following parameters:

-   **N**: number of indexed elements.
-   **x**: input [`Array`][mdn-array] or [`typed array`][mdn-typed-array].
-   **strideX**: stride length for `x`.

> The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to calculate the count of every other element in `x`,

Reviewer: ```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to calculate the count for every other element in `x`,
```

```javascript
var x = [ 1.0, 2.0, 2.0, -7.0, -2.0, 3.0, 4.0, 2.0, NaN, NaN ];

var v = nancount( 5, x, 2 );
// returns 4
```

Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.

<!-- eslint-disable stdlib/capitalized-comments -->

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var x0 = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, NaN, NaN, 4.0 ] );
var x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element

var v = nancount( 4, x1, 2 );
// returns 3
```

#### nancount.ndarray( N, x, strideX, offsetX )

> Calculate the number of non-`NaN` elements in a  strided array

Reviewer: ```suggestion
Calculates the number of non-`NaN` elements in a strided array using alternative indexing semantics.
```
